.github/workflows/tests-unit-php.yml

```diff
@@ -116,15 +115,29 @@ jobs:
         with:
           composer-options: '--prefer-dist --no-progress --no-interaction --no-scripts'
 
-      # Installs PHPUnit 9.5+ for tests against trunk and PHPUnit 7.5.x for older versions,
-      # because WordPress 5.8 and below don't support newer PHPUnit versions.
-      # PHPUnit 7.5 does support PHP 7.0 though, so we don't run an update there.
+      # Installs a different PHPUnit version depending on the WP/PHP version combo we're testing against.
+      #
+      # | WP  / PHP | PHPUnit |
+      # |-----------|---------|
+      # | 5.5 / 7.2 | 7       |
+      # | 5.9 / 7.2 | 8       |
+      # | 5.9 / 7.4 | 9       |
+      # | *   / 8   | 9       |
+      #
+      # See https://make.wordpress.org/core/handbook/references/phpunit-compatibility-and-wordpress-versions/
       - name: Update PHPUnit
         run: |
-          if [[ $WP_VERSION == "trunk" ]]; then
-            composer update --ignore-platform-reqs --no-interaction --no-scripts yoast/phpunit-polyfills --with-dependencies
-          elif [[ $PHP_VERSION != "7.0" ]]; then
+          if [[ $WP_VERSION == "5.5" && $PHP_VERSION == "7.2" ]]; then
+            echo "Installing PHPUnit 7.5.x"
             composer require --ignore-platform-reqs --no-interaction --no-scripts phpunit/phpunit:^7.5 --with-dependencies
+          elif [[ $PHP_VERSION == "7.2" ]]; then
+            echo "Installing PHPUnit 8.5.x"
+            composer require --ignore-platform-reqs --no-interaction --no-scripts phpunit/phpunit:^8.5 --with-dependencies
+            echo "Downgrading phpunit/php-token-stream because the latest version requires PHP 7.3"
+            composer require --ignore-platform-reqs --no-interaction --no-scripts phpunit/php-token-stream:^3.1.3 --with-dependencies
+          elif [[ $WP_VERSION == "latest" || $WP_VERSION == "trunk" || $PHP_VERSION == "7.4" ]]; then
+            echo "Installing latest version of PHPUnit"
+            composer update --ignore-platform-reqs --no-interaction --no-scripts yoast/phpunit-polyfills --with-dependencies
```

Comment:
Can you explain what is happening here?

Reply:
The goal is to install the latest possible PHPUnit version for a given WP / PHP version (see table + link above).

Since `yoast/phpunit-polyfills` has [PHPUnit as its dependency](https://github.com/Yoast/PHPUnit-Polyfills/blob/5cd00434959285ac65b90071d88345d514c6b06a/composer.json#L27), this line here simply means "Update to whatever is the latest PHPUnit supported by yoast/phpunit-polyfills"

This is currently PHPUnit 9, the latest version.

Since this line is only called for WP latest / WP trunk / PHP 7.4, which all support PHPUnit 9, this means these tests are now running with PHPUnit 9.

You can check this by going through the GitHub action log to see which version is being used.

Here's an example: https://github.com/GoogleForCreators/web-stories-wp/runs/4980454198?check_suite_focus=true#step:15:10

Note that this line and this logic already existed before, I just moved it around.